### PR TITLE
Setup jscs to enforce code style

### LINKED ISSUE #342: Whitespace adjust grunt task or git hook
Add a grunt task or tasks (or git hook) to apply or revert 2 space indentation.

### PR DESCRIPTION
**Pull request by taylortom**
Fixes #342.

Have based the [jscs settings](https://github.com/adaptlearning/adapt_authoring/blob/issue/342/.jscsrc) on a node-inspired configuration. I've set the task to auto-enforce the rules where possible, so we may want to take some of the options out. 

See [here](http://jscs.info/rules) for the full list of available options.

### CODE REVIEW

**Comment by taylortom** [comment]
There's also a git hook plugin that can run this task before commit

**Comment by brian-learningpool** [comment]
+1. This throws 240 warnings on the current codebase right now.  Anyone fancy tidying them up?

**Comment by dennis-learningpool** [comment]
+1

File: .jscsrc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "requireSpaceAfterLineComment": true,
+  "requireSpaceAfterBinaryOperators": true,
+  "requireSpaceBeforeBinaryOperators": true,
+  "requireSpaceBeforeBlockStatements": true,
+  "requireSpaceBeforeObjectValues": true,
+  "requireSpacesInFunction": {
+    "beforeOpeningCurlyBrace": true
+  },
+  "requireTrailingComma": {
+    "ignoreSingleLine": true
```
Comment (dennis-learningpool) [inline .jscsrc]: does this mean adding a comma after the final entry in an array? this can be dodgy in front end JS, i think.

Comment (brian-learningpool) [inline .jscsrc]: Perhaps we should use `"disallowTrailingComma": true` instead?

**Comment by lc-thomasberger** [comment]
+1

**Pull request merged**